### ISSUE TITLE
Where ImageNet images resized such that the smaller dimension is 256? 

### ISSUE THREAD

**Issue by JosephKJ** [yaoyao-liu/class-incremental-learning#27]
Hi @yaoyao-liu,

One more clarification request, @yaoyao-liu. 

Where ImageNet images resized such that the smaller dimension is 256? It would be great if you could recall this. 

Thanks,
Joseph

**Comment by yaoyao-liu** [comment]
Maybe you mean this:
https://github.com/yaoyao-liu/class-incremental-learning/blob/fe9a7bee0bf46d69f70373934feb87a8f7134d4b/adaptive-aggregation-networks/trainer/base_trainer.py#L125-L130

**Comment by JosephKJ** [comment]
A bit different though. I was running the ImageNet-Subset experiment. I explicitly resized all training images such that the smaller dimension is 256 following [this](https://github.com/soumith/imagenet-multiGPU.torch/blob/deb5466a16e54ec7a69fe027e5fbcd3c1bfb49cc/README.md) (hoping that the data-loader might be a bit faster).

This is the exact command:
`find . -name "*.JPEG" | xargs -I {} convert {} -resize "256^>" {}
`

So essentially in base_trainer.py we RandomResizedCrop images to 224, 
https://github.com/yaoyao-liu/class-incremental-learning/blob/fe9a7bee0bf46d69f70373934feb87a8f7134d4b/adaptive-aggregation-networks/trainer/base_trainer.py#L123-L124

I do get the following numbers on ImageNet-subset 5 task setting for iCaRL:

Computing accuracy on the 0-th phase classes...
  Current accuracy (FC)         :               27.48 %
  Current accuracy (Proto)      :               44.08 %
  Current accuracy (Proto-UB)   :               58.24 %

Computing cumulative accuracy...
  Current accuracy (FC)         :               42.74 %
  Current accuracy (Proto)      :               56.40 %
  Current accuracy (Proto-UB)   :               62.34 %

Computing average accuracy...
  Average accuracy (FC)         :               58.24 %
  Average accuracy (Proto)      :               67.67 %

The paper reports: 65.44, and we get 58.24 (I believe you are reporting FC results for iCaRL too, please correct me if I am wrong)

These are the hyper-params that I changed for ImageNet experiments (when compared to the CIFAR experiments) to match LUCIR's code base as you suggested in this repo:  
```
--epochs 90 # It was 160 for CIFAR experiments
--custom_weight_decay 1e-4 # It was 5e-4 for CIFAR experiments
--num_workers 16 # It was 8 for CIFAR experiments
```

I am just wondering whether we have the drop in accuracy because I resized the training images. Would be great to have your thoughts on this.

**Comment by yaoyao-liu** [comment]
If you are running the experiments for the baseline iCaRL, it seems that the results are correct.
For iCaRL, the result is denoted as `Average accuracy (Proto)`. It is 67.67%. 
FC results are not for iCaRL. In iCaRL, the authors only use the prototype classifier. You may refer to the original paper: [\[Link\]](https://arxiv.org/abs/1611.07725).

**Comment by JosephKJ** [comment]
Got it, thanks @yaoyao-liu
